### PR TITLE
helm: Add additional labels to charts

### PR DESCRIPTION
**Pull request by davidjumani**
Continuation of https://github.com/solo-io/gloo/pull/9195

# Description
This new feature is to add additionalLabels to helm chart

# Context
See issue [here](https://github.com/solo-io/gloo/issues/9035)

## Testing steps
I manually verified behavior by templating the helm chart with different values

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Review by Rachael-Graham — DISMISSED** [review]

File: install/test/helm_test.go [hunk 1]
```diff
@@ -6757,6 +6767,21 @@ metadata:
 				Expect(resources.NumResources()).To(Equal(2))
 			})
 
+			// As every resource created should have the additional labels, testing everyone of them can be tedious.
+			// Instead just validating that one of them has it is sufficient to ensure that the chart renders with the additional labels
```
Comment (davidjumani) [inline install/test/helm_test.go:6771]: Any thoughts on this ?

Comment (sheidkamp) [reply install/test/helm_test.go:6771]: I think there's 2 things to test:
* that the include template renders correctly
* that the template is included everywhere it need to be

This test addresses the first point. I agree that tests in this format for the second point would be cumbersome. Maybe we can address this with some kind of grep/shell command that can look for manual applications of `app: gloo` and `labels` that don't use the include? The command could then be added to the PR description and this comment for later reference.

File: install/helm/gloo/generate/values.go [hunk 2]
```diff
@@ -38,6 +38,7 @@ type Global struct {
 	Graphql              interface{}        `json:"graphql,omitempty" desc:"(Enterprise Only): GraphQL configuration options."`
 	ConfigMaps           []*GlobalConfigMap `json:"configMaps,omitempty" desc:"Config used to create ConfigMaps at install time to store arbitrary data."`
 	ExtraCustomResources *bool              `json:"extraCustomResources,omitempty" desc:"Add additional custom resources to create, as defined by a helm partial. Defaults to false in open source, and true in enterprise."`
+	AdditionalLabels     map[string]string  `json:"additionalLabels,omitempty" desc:"Additional labels to add to gloo resources."`
```
Comment (sheidkamp) [inline install/helm/gloo/generate/values.go:41]:
```suggestion
	AdditionalLabels     map[string]string  `json:"additionalLabels,omitempty" desc:"Additional labels to add to all gloo resources."`
```

**Review by sheidkamp — COMMENTED** [review]
Some thoughts on the tests

File: install/test/helm_test.go (hunk 1 above)
Comment (davidjumani) [reply install/test/helm_test.go]: Thanks. Found a way to expect the labels on all resources

Comment (sheidkamp) [reply install/test/helm_test.go]: I like this test, but it only covers the resources that happen to get rendered with the default helm.

Can we still add command line type of check to examine every template?

**Review by sheidkamp — CHANGES_REQUESTED** [review]

File: install/test/helm_test.go (hunk 1 above)
Comment (davidjumani) [reply install/test/helm_test.go]: The output of `helm template` is converted and checked for the required labels (just like a grep)
I've added custom helm settings to enable components that generate their own resources

Comment (sheidkamp) [reply install/test/helm_test.go]: Even better

**Review by sheidkamp — APPROVED** [review]

**Pull request merged**